HoldingPen: remove duplicates from categories filter

The [categories filter](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/static/js/inspire_workflows_ui/holdingpen/holdingpen.filters.js#L55-L81) should remove duplicates.

Otherwise, for example creating a submission with the arXiv ID `hep-th/9711200` will trigger the following error in the Holding Pen results page:

```
inspire-workflows.details.659ac3b8.js:12811 Error: [ngRepeat:dupes] Duplicates in a repeater are not allowed. Use 'track by' expression to specify unique keys. Repeater: category in record._source.metadata | categories, Duplicate key: string:Theory-HEP, Duplicate value: Theory-HEP
```

--- a/inspirehep/modules/workflows/static/js/inspire_workflows_ui/holdingpen/holdingpen.filters.js
+++ b/inspirehep/modules/workflows/static/js/inspire_workflows_ui/holdingpen/holdingpen.filters.js
@@ -78,7 +78,10 @@
         }
       }
 
-      return categories;
+      var uniqueCategories = categories.filter(function (value, index, self) {
+        return self.indexOf(value) === index;
+      });
+      return uniqueCategories;
     }
   }
 

